\documentclass{bioinfo}
\copyrightyear{2012}
\pubyear{2012}

\begin{document}
\firstpage{1}

\title[comb-p]{Comb-p: software for combining, analyzing, grouping, and correcting spatially correlated p-values}
\author[Pedersen \textit{et~al}]{Brent S. Pedersen\,$^{1,*}$, Katerina J. Kechris\,$^{2}$,
    Ivana V. Yang\,$^{1}$ and David A. Schwartz\,$^1$}
\address{$^{1}$Department of Medicine, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
$^{2}$Department of Statistics, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
}
\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}
\maketitle
\begin{abstract}

\section{Summary:}
\textit{comb-p} is command-line tool and a python library that
manipulates BED files of p-values and 1) calculates autocorrelation, 2) Combines
adjacent p-values based on the Stouffer-Liptak-Kechris correction \citep{Kechris2010},
3) performs false discovery correction, 4) finds troughs of adjacent low p-values,
and 5) assigns significance to those regions.
In addition, a number of tools are provided for visualization and
assessment. The library sets sensible defaults for parameters at each
step based on the structure of the input data, but these may also be customized
by the user. We provide example uses on bisulfite-seq, ChIP-seq and tiled methylation
probes.

\section{Availability:}
\textit{comb-p} is maintained under the BSD license. The documentation and
implementation are available at

\href{https://github.com/brentp/combined-pvalues}{https://github.com/brentp/combined-pvalues}.
\section{Contact:} \href{bpederse@gmail.com}{bpederse@gmail.com}
\end{abstract}

\section{Introduction}
ChIP-seq, bisulfite-seq, and tiling arrays generate data that is spatially
auto-correlated. The values from these and many other technologies can be
compared or tested to generate p-values that are also auto-correlated. Due to the
high-throughput nature of these methods, the significance of individual regions
may be obscured after multiple-testing correction on potentially millions of sites.
Here, we build on the approach of combining p-values described in \citep{Kechris2010}.

\section{Approach}

\citep{Fisher} developed an approach of combining p-values from independent tests
to get a single meta-analysis test-statistic with a chi-squared distribution 
and degrees of freedom based on the number of tests being combined.
A similar method developed by \citep{Stouffer} first converts p-values
to Z-scores which are then summed and scaled to create a single, combined Z-score.
Stouffer's method lends itself well to the addition of weights on each p-value
\citep{Liptak}.

\citep{Zaykin} extended the weighting explicitly to deal with auto-correlation and
\citep{Kechris2010} used that framework to create a sliding window correction where
the p-value of each probe is adjusted by applying Stouffer's method to neighboring
p-values as weighted according to the observed auto-correlation at the appropriate
distance.


\begin{methods}

\section{Implementation}

\textit{comb-p} is implemented as a single command-line application that
dispatches to multiple independent sub-modules, and as a set of python
packages that may be used programatically. Where possible,
computationally-intensive parts of the algorithm are parallelized
automatically--for the ACF and the SLK steps described below--this
results in a speedup that is linear with the number of cores available.
In all cases the implementation has been tuned to minimize memory-use and
computation time. All programs expect files simple BED format \citep{TODO-UCSC?}
sorted by chromosome and start.
When run without arguments, the \textit{comb-p} executable displays the
available programs and a short description of each:
\begin{verbatim}
acf      - autocorrelation within BED file
slk      - Stouffer-Liptak-Kechris correction
fdr      - Benjamini-Hochberg correction
peaks    - find peaks in a BED file.
region_p - SLK p-values for a region
hist     - histogram of a p-values
splot    - scatter plot of a region.
manhattan - a manhattan plot
pipeline  - slk, fdr, peaks, region_p
\end{verbatim}
Though each of these tools may be used independently, the progression of
\textit{acf}, \textit{slk}, \textit{fdr}, \textit{peaks}, \textit{region\_p}
described below works well for our data [Supplemental]. These steps can
be run successively with the single command: \textit{pipeline}.

Given a set of auto-correlated p-values in BED format, \textit{comb-p} first
calculates the correlation at varying distance lags. Whereas \citep{Kechris2010}
and many ACF implementations rely on fixed offsets of adjacent probes,
\textit{comb-p} accepts a maxiumum base-pair distance and bins pairs of probes
based on their distance, so a given probe may have multiple probes in each of
a set of bins. This is useful in cases where the probes generating the p-values
are unevenly spaced. We refer to this set of lagged correlations as the ACF for
auto-correlation function.

Once the ACF has been calculated, it can be used to perform the
Stouffer-Liptak-Kechris correction (\textit{slk}) where each p-value is
adjusted according to the p-values in its neighborhood and the weight applied
to each probe that neighborhood according to the ACF. The resulting BED file
has the same number of rows with an additional column containing the corrected
p-value. A given p-value will be pulled lower if it's neighbors also have low
p-values and likely remain insignificant if the neighboring p-values are also 
high.

A q-value score based on the Benjamini-Hochberg false-discovery (FDR) correction
 may be calculated. A peak-finding algorithm can then be used on either the FDR
q-value, the \textit{slk}-corrected p-value, or even the original, uncorrected 
p-value. We find that performing the \textit{slk} correction and the FDR correction
before finding regions is the most stable and powerful.

A p-value for each region can be assigned using the Stouffer-Liptak correction.
This step varies from the \textit{slk} step in that the \textit{slk} step uses
the ACF calculated out to a fixed lag where as the \textit{region\_p} step first
calculates the ACF out to a distance equal to the largest region-size. The
corrected p-value for each region is then calculated using the original p-values that
fall within the region, and the portion of the ACF that extends to the distance
spanned by that region. Because we use the original, uncorrected p-values in the
calculation of significance for the region, we side-step issues of altering the
distribution in both the \textit{slk} and FDR steps. The \textit{region\_p}
program reports the \textit{slk} corrected p-value, and a further one-step
Sidak \citep{Sidak}
multiple-testing correction. The number of possible tests in the Sidak correction
is the total bases covered by the input probes divided by the extent of the
given region. In our simulations, this correction gives a more strigent p-value
than simulations [Supplemental?].
TODO: single plot with subplot for ACF, region in UCSC.

\section{Application}
Comprehensive, High-throughput Array for Relative Methylation or CHARM is a
tiling array used to quantify methylation at CpG-rich sites \citep{Irizarry}.
Adjacent probes are spatially auto-correlated due to the regional nature of
methylation, and to the overlapping probes on the tiling array. In the
Supplemental Info, we show the commands used to run \textit{comp-p} on data from
??? . In that analysis, we find several interesting regions, including those
associated with ??? [TODO: cite Ivana's LGRC/MRCA paper].

Bisulfite-sequencing data is also used to uncover methylation. We analyze
data from {\it Arabidopsis thaliana}, comparing ??? to ??? TODO and find
that genes associated with ??? are hyper/hypo-methylated. This observation
was previously reported in ??? TODO.

ChIP-seq data is also known to be spatially autocorrelated. We analyzed data
from the transcription factor ???, comparing ??? to ??? TODO, and find
???

\end{methods}

\section{Conclusion}
The \textit{comb-p} software is useful in contexts where there is
autocorrelation among nearby p-values. We have outlined our implementation
and demonstrated the utility on data from three different technologies.

\section*{Acknowledgement}

\paragraph{Funding\textcolon} This work was supported by ??? to ???.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}

%\bibliographystyle{plain}

%\bibliography{Document}
\begin{thebibliography}{}
\bibitem[Kechris {\it et~al}., 2010]{Kechris2010} Kechris, K.J. et al. (2010)
Generalizing moving averages for tiling arrays using combined p-value
statistic. {\it Statistical Applications in Genetics and Molecular Biology}
{\bf 9}, Article 29.
\bibitem[Quinlan and Hall, 2010]{Quinlan2010} Quinlan, A.R. and Hall, I.M. (2010) BEDTools: a flexible suite of utilities for comparing genomic features, {\it Bioinformatics}, {\bf 26}, 841-842.
\bibitem[Kent \textit{et al.}, 2002]{Kent2002} Kent, W.J. et al. (2002) The human genome browser at UCSC. {\it Genome Res.}, {\bf 12}, 996-1106.

\bibitem[Sidak 1967]{Sidak} 
Šidàk, Z. (1967).
Rectangular conﬁdence region for the means of multivariate normal distributions.
\textit{Journal of the American Statistical Association}, {\bf 62}, 626-633.

\bibitem[Fisher 1948]{Fisher} 
Fisher, R.A. (1948)
Questions and answers \#14.
{\it The American Statistician}, {\bf 2}(5), 30-31.

\bibitem[Stouffer \textit{et al.} 1949]{Stouffer}
Stouffer, S.A. \textit{et~al.} (1949). \textit{The American Soldier},
Princeton University Press, Princeton, NJ. Vol.1: Adjustment during Army Life.

\bibitem[Liptak 1958]{Liptak}
Liptak, T. (1958). On the combination of independent tests. {\it Magyar Tud. Akad. Mat. Kutato Int. Kozl.}, {bf 3}, 171-197.

\bibitem[Zaykin Year]{Zaykin} Zaykin TODO
\bibitem[Irizarry Year]{Irizarry} Irizarry TODO
\bibitem[Pedersen Year]{Pedersen} Pedersen TODO

\end{thebibliography}
\end{document}
