\documentclass{bioinfo}
\copyrightyear{2012}
\pubyear{2012}

\begin{document}
\firstpage{1}

\title[comb-p]{Comb-p: software for combining, analyzing, grouping, and correcting spatially correlated p-values}
\author[Pedersen \textit{et~al}]{Brent S. Pedersen\,$^1$\footnote{to whom correspondence should be addressed},
 David A. Schwartz\,$^1$,
    Ivana V. Yang\,$^1$\footnote{The authors wish it to be known that in their opinion,
the last two authors should be regarded as joint Last Authors.}

and Katerina J.  Kechris\,$^2$\footnotemark[2]{}

}

\address{$^{1}$Department of Medicine, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045,
USA\\
$^{2}$Department of Biostatistics and Informatics, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
}


\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}
\maketitle
\begin{abstract}

\section{Summary:}
\textit{comb-p} is a command-line tool and a python library that
manipulates BED files of possibly irregularly spaced p-values and
1) calculates auto-correlation,
2) combines adjacent p-values,
3) performs false discovery adjustment,
4) finds regions of enrichment (i.e., series of adjacent low p-values), and
5) assigns significance to those regions.
In addition, tools are provided for visualization and
assessment.
We provide validation and example uses on bisulfite-seq with p-values
from Fisher's exact test,
tiled methylation probes using a linear model, and Dam-ID for chromatin binding
using moderated t-statistics. Because the library
accepts input in a simple, standardized format and is unaffected by the origin
of the p-values, it can be used for a wide variety of applications.

\section{Availability:}
\textit{comb-p} is maintained under the BSD license. The documentation and
implementation are available at

\href{https://github.com/brentp/combined-pvalues}{https://github.com/brentp/combined-pvalues} .

\section{Contact:} \href{bpederse@gmail.com}{bpederse@gmail.com}

\end{abstract}

\section{Introduction}
A variety of high-throughput technologies generate
genome-wide data used to study processes such as DNA-binding, methylation
status and histone modifications. These technologies, including tiling
arrays and sequence-based assays generate data
that are often auto-correlated across
the genome, making inferences difficult. The significance of
individual regions may be dampened after multiple-testing correction
on potentially millions of sites.
In such studies, hypotheses tests can be performed at each location to
generate p-values for evaluating the effects of interest. For this
purpose, \cite{Kechris2010} developed a method for combining p-values
in sliding windows and accounting for spatial correlations across the
genome. Here, we build on this approach with software
that allows for uneven data structure across the genome, more general
auto-correlation calculations and multiple-testing corrections for peaks
(i.e., genomic regions of enrichment) with applications to a variety of
different technologies.

\section{Approach}
Tiling array studies relying on two-sample comparisons may be
amenable to the calculation of sliding window averages of log ratios,
or two-sample test statistics. However,
more complex study designs often require covariates and report
p-values from linear models or other statistical tests.

We utilize a 'moving averages' method of p-value correction that
does not depend on the test used to generate the p-values. \cite{Fisher}
developed an approach of combining p-values from independent tests to
get a single meta-analysis test-statistic with a chi-squared
distribution and degrees of freedom based on the number of tests being
combined.  A similar method developed by \cite{Stouffer} and
\cite{Liptak} first converts p-values to Z-scores which are then summed
and scaled to create a single, combined Z-score.  The Stouffer-Liptak
method lends itself to the addition of weights on each p-value.
\cite{Zaykin2002} introduced a method to use weights to perform a
dependence correction on correlated tests. \cite{Kechris2010} used
a sliding window correction where each p-value is
adjusted by applying the Stouffer-Liptak method to neighboring p-values
as weighted according to the observed auto-correlation at the
appropriate lag.

Here, we provide a generic, efficient, and customizable implementation
of these methods with additions including handling variably spaced
probes, a peak finder for dynamically sized regions, and a means to
calculate a p-value for each peak, adjusted for multiple
comparisons. We refer to this implementation as \textit{comb-p} and
illustrate its application using three different technologies.

\begin{methods}

\section{Implementation}

All programs within \textit{comb-p} expect files in simple BED format
\citep{Kent2002} sorted by chromosome and start.
Additional columns contain the p-value(s) of interest based on the
study design and generated from any software or statistical test.


\textit{comb-p} first
calculates the correlation at varying distance lags
(here: auto-correlation - ACF).
 Whereas Kechris'
\citep{Kechris2010} and many ACF implementations rely on fixed offsets
of adjacent probes, \textit{comb-p} accepts a maximum
distance and a single offset that segments that distance into intervals.
If a given probe is 40 bases away from two (or more) other probes, then it will
appear more than once in the bin containing probe-pairs separated by less
than 40 bases. This is useful in cases where the probes generating the p-values
are unevenly spaced, as is common with methylation arrays.

Once the ACF has been calculated, it can be used to perform the
Stouffer-Liptak-Kechris correction (\textit{slk}) where each p-value is
adjusted according to adjacent p-values as weighted
according to the ACF. The resulting BED file
has an additional column containing the corrected p-value.
A given p-value will be pulled lower if its neighbors also have low
p-values (and little auto-correlation) and likely remain insignificant if the
neighboring p-values are also high.

A q-value score based on the Benjamini-Hochberg false-discovery (FDR)
correction or on a null-model from shuffled data may then be calculated.
The peak-finding algorithm can then be
used to find enrichment regions or \textit{peaks} on either the FDR
q-value, the \textit{slk}-corrected p-value, or on the original
p-value.

Once the regions are identified, a p-value for each region can be assigned
using the Stouffer-Liptak correction.
Note that the first ACF calculated
above was only out to the distance specified and may not extend far enough
to cover the longer regions.
Therefore, this step first calculates the ACF out to a distance equal
to the length of the longest region. The corrected
p-value for each region is then calculated using the original p-values
that fall within the peak and the full ACF.
Because we use the original,
uncorrected p-values in the calculation of significance for the peak,
we side-step issues of altering the distribution in both the
\textit{slk} and FDR steps. The \textit{region\_p} program reports the
\textit{slk} corrected p-value, and a one-step \cite{Sidak}
multiple-testing correction.
For a given region, the number of possible tests
in the \v{S}id\'{a}k correction is the total bases covered by all input probes
divided by the size of the given region.
In short, we define the extents of the region using the FDR q-values,
then we define the significance of the regions using the SLK correction
of the original p-values.

\textit{comb-p} is implemented as a single command-line application that
dispatches to multiple independent sub-modules, and as a set of python
packages that may be used programatically. Where possible,
computationally-intensive algorithms are parallelized
automatically--for the ACF and the \textit{slk} steps in the analyses
described below--this
results in a speed-up that is nearly linear with the number of cores available.
The implementation has been tuned to minimize memory use and
computation time.

When run without arguments, the \textit{comb-p} executable displays the
available programs and a short description of each.
Though each of these tools may be used independently, the progression of
\textit{acf}, \textit{slk}, \textit{fdr}, \textit{peaks}, \textit{region\_p}
works well for the examples presented. These steps can
be run successively with the single command: \textit{pipeline}.

\section{Application}
We extend 3 previously published analyses to demonstrate the utility and
breadth of use of \textit{comb-p} and to validate against the
published data. The full set of commands used to run each of these analyses
is available in the corresponding subdirectory at:
https://github.com/brentp/combined-pvalues/tree/master/examples

Comprehensive, High-throughput Array for Relative Methylation (CHARM)
is a tiling array used to quantify methylation at CpG-rich sites
\citep{Irizarry2008}.  Adjacent probes are correlated
due to the regional nature of methylation and to the overlapping
probes on the tiling array.  The CHARM study reported in
\cite{Irizarry2009} contains probe data for tumor and normal samples
from a variety of tissues. Using these data, we fit a linear model in R
(http://www.R-project.org) to
obtain p-values for the significance of tumor or tissue-specific
effects on methylation status at each probe.  We then run
\textit{comb-p}, which finds 85\% of the tumor-specific differentially
methylated regions (C-DMRs) reported by \cite{Irizarry2009} and 75\% of
the tissue-specific T-DMRs reported.  While these overlaps depend on
the parameter-choice, we show that the Irizarry DMRs that overlap
with the \textit{comb-p} DMRs have signficantly (t-test,
p=1.296e-145) lower FDR values (as reported by Irizarry) than those
that do not, indicating that \textit{comb-p} is reporting the best regions.
We also show that our method yields only 37 false-positive DMRs on 20
runs of shuffled of CHARM data (33 of those come from 2 of the sets,
because those shuffles of the clinical data happen to coincide well
with the case-control status of the original data). This gives a false
discovery rate of 0.0016.
We found that \textit{comb-p} recovers more significant
probes than a simple FDR correction on the original p-values.
We also found that the process of segmenting the genome into smaller chunks and
calling regions independently in each chunk generates similar regions to those
derived using the autocorrelation structure for the entire genome (the default).


DamID technology followed by tiling arrays can be used to map genomic
regions bound by DNA-binding proteins \citep{Steensel2001}.  We
recreate an analysis of the DamID tiling array data for the Ci
transcription factor in {\it Drosophila melanogaster} \citep{Biehs}.
\cite{Kechris2010} calculated the
mean log ratios of intensities between Ci experimental and control
samples to obtain p-values from a one-sided moderated t-test.
Of the original 878 regions, 695 (79\%) are
represented in the \textit{comb-p} predicted peaks.  To score each
peak, the authors report $-\log_{10}$ of the smallest raw p-value for
each peak. The published peaks overlappng the \textit{comb-p} regions
have a higher score than those that do not (t-test, p=2.98e-59),
indicating that \textit{comb-p}
is finding the best regions among those previously reported.
In addition, \textit{comb-p} regions also yield a higher target enrichment
score based on proximity to known Ci genes, indicating that \textit{comb-p}
is selecting for known targets.

Bisulfite-sequencing (BS-Seq) is also used to measure methylation across the
genome. As another example of the flexibility of our method, we demonstrate
a possible analysis on data described in \cite{Hsieh2009} from {\it Arabidopsis
thaliana} using MethylCoder \citep{Pedersen2011} to map the
bisulfite-treated reads to the genome. At each site,
we use Fisher's exact test to obtain p-values for the counts of
converted and un-converted cytosines between endosperm and embryo. We find
differentially methylated regions between these two tissues associated
with genes enriched for gene ontologies related to the ribosome (p=
1e-3).
\end{methods}

\section{Conclusion}

The \textit{comb-p} software is useful in contexts where auto-correlated
p-values are generated across the genome.
We have outlined our implementation and demonstrated
the utility on data from three different technologies each from a
different statistical test. We have also validated our method using
previously published results from those studies.
Future work could extend this framework to address applications
with a more segmented structure such as DNA copy-number and ChIP-seq.
Currently, we recommend the use of \textit{comb-p} for technologies with a
more regular autocorrelation structure.

\section*{Acknowledgement}
\paragraph{Funding\textcolon}
This work was supported by the National Institute of Health grants
AA016922 to KK, HL101251 to DAS and IVY and AI90052 and HL101715 to DAS.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%
%\bibliographystyle{plain}
%
%\bibliography{Document}

\begin{thebibliography}{}
\bibitem[Biehs {\it et~al}, 2010]{Biehs}
Biehs, B., et al. (2010)
Hedgehog targets in the Drosophila embryo and the mechanisms that generate tissue-specific outputs of Hedgehog signaling. {\it Development}, {\bf 137}, 3887-3898.

\bibitem[Fisher, 1948]{Fisher}
Fisher, R.A. (1948)
Questions and answers \#14.
{\it The American Statistician}, {\bf 2}(5), 30-31.

\bibitem[Hsieh {\it et~al}., 2009]{Hsieh2009} Hsieh, T.F. et al. (2009)
Genome-Wide Demethylation of Arabidopsis Endosperm.
{\it Science}, {\bf 324}, 1451-1454.

\bibitem[Irizarry {\it et~al}., 2008]{Irizarry2008} Irizarry, R.A. et al.
(2008) Comprehensive high-throughput arrays for relative methylation (CHARM).
{\it Genome Research}, {\bf 18}, 780-790.

\bibitem[Irizarry {\it et~al}., 2009]{Irizarry2009} Irizarry, R.A. et al.
(2009) The human colon cancer methylome shows similar hypo- and
hypermethylation at conserved tissue-specific CpG island shores.
{\it Nature Genetics}, {\bf 41}, 178-186.

\bibitem[Kechris {\it et~al}., 2010]{Kechris2010}
Kechris, K.J. et al. (2010)
Generalizing moving averages for tiling arrays using combined p-value
statistic. {\it Statistical Applications in Genetics and Molecular Biology}
{\bf 9}, Article 29.

\bibitem[Kent {\it et~al}, 2002]{Kent2002} Kent, W.J. et al. (2002) The human genome browser at UCSC. {\it
Genome Res.}, {\bf 12}, 996-1106.

\bibitem[Liptak, 1958]{Liptak}
Liptak, T. (1958) On the combination of independent tests.
{\it Magyar Tudomanyos. Akademia Matematikai Kutato Intezetenek Kozlemenyei}, {\bf 3}, 171-197.

\bibitem[Pedersen {\it et~al}., 2011]{Pedersen2011} Pedersen, B.S. et al.
(2011) MethylCoder: software pipeline for bisulfite-treated sequences.
{\it Bioinformatics}, {\bf 27}, 2435-2436.

\bibitem[\v{S}id\'{a}k, 1967]{Sidak}
\v{S}id\'{a}k, Z. (1967)
Rectangular confidence region for the means
 of multivariate normal distributions.
\textit{Journal of the American Statistical Association}, {\bf 62}, 626-633.

\bibitem[van Steensel {\it et~al}., 2001]{Steensel2001}van Steensel, B., Delrow, J. and Henikoff, S. (2001)
Chromatin Profiling Using Targeted DNA Adenine Methyltransferase,
{\it Nature Genetics}, {\bf 27}, 304-308.

\bibitem[Stouffer \textit{et~al.}, 1949]{Stouffer}
Stouffer, S.A. \textit{et~al.} (1949) \textit{The American Soldier},
Princeton University Press, Princeton, NJ. Vol.1: Adjustment during Army Life.

\bibitem[Zaykin {\it et~al}., 2002]{Zaykin2002} Zaykin, D.L. et al. (2002)
Truncated product method for combining p-values. {\it Genetic Epidemiology},
{\bf 22}, 170-185.

\end{thebibliography}
\end{document}
