\documentclass{bioinfo}
\copyrightyear{2012}
\pubyear{2012}

\begin{document}
\firstpage{1}

\title[comb-p]{Comb-p: software for combining, analyzing, grouping, and correcting spatially correlated p-values}
\author[Pedersen \textit{et~al}]{Brent S. Pedersen\,$^{1,*}$,
Katerina J.  Kechris\,$^{2,*}$,
    Ivana V. Yang\,$^{1}$ and David A. Schwartz\,$^1$}
\address{$^{1}$Department of Medicine, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
$^{2}$Department of Statistics, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
}
\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}
\maketitle
\begin{abstract}

\section{Summary:}
\textit{comb-p} is command-line tool and a python library that
manipulates BED files of possibly irregularly spaced p-values and
1) calculates autocorrelation,
2) Combines adjacent p-values based on the Stouffer-Liptak-Kechris correction \citep{Kechris2010},
3) performs false discovery correction,
4) finds troughs of adjacent low p-values, and
5) assigns significance to those regions.
In addition, tools are provided for visualization and
assessment. The library sets sensible defaults for parameters at each
step based on the structure of the input data, but these may also be customized
by the user. We provide validation and example uses on bisulfite-seq with p-values
from fisher's exact test,
tiled methylation probes using a linear model, and Dam-ID for chromatin binding
using t-statistics [TODO Katerina, correct?]. Because the library
accepts input in a simple, standardized format and is unaffected by the origin
of the p-values, it can be used for a wide variety of applications.

\section{Availability:}
\textit{comb-p} is maintained under the BSD license. The documentation and
implementation are available at

\href{https://github.com/brentp/combined-pvalues}{https://github.com/brentp/combined-pvalues} .
Descriptions of examples and the scripts used to run them are at:

\href{https://github.com/brentp/combined-pvalues/tree/master/examples/}{https://github.com/brentp/combined-pvalues/tree/master/examples/} .

\section{Contact:} \href{bpederse@gmail.com}{bpederse@gmail.com},
\href{katerina.kechris@ucdenver.edu}{katerina.kechris@ucdenver.edu}

\end{abstract}

\section{Introduction}
Methylation data and data from tiling arrays generate spatially auto-correlated
values. Such data can be compared or tested to generate p-values that are also
auto-correlated. Due to the high-throughput nature of these methods, the
significance of individual regions may be dampened after multiple-testing correction
on potentially millions of sites. Here, we build on the approach of combining
p-values described in \cite{Kechris2010} that mitigates this loss by
accounting for autocorrelation.
\section{Approach}
\cite{Fisher} developed an approach of combining p-values from independent tests
to get a single meta-analysis test-statistic with a chi-squared distribution
and degrees of freedom based on the number of tests being combined.
A similar method developed by \cite{Stouffer} first converts p-values
to Z-scores which are then summed and scaled to create a single, combined Z-score.
Stouffer's method lends itself well to the addition of weights on each p-value
\citep{Liptak}.

\cite{Zaykin2002} introduced a method to use weights to perform dependence
correction on correlated tests. \cite{Kechris2010} used that framework to create
a sliding window correction where each p-value is adjusted by
applying Stouffer's method to neighboring p-values as weighted according to the
observed auto-correlation at the appropriate distance.

Here, we by provide a generic, efficient, and customizable implementation
of these methods with additions including handling variably spaced probes,
a peak finder for dynamically sized regions, and a means to calculate a
p-value for each peak. We refer to this implemenation as {\textit comb-p}.

\begin{methods}

\section{Implementation}

\textit{comb-p} is implemented as a single command-line application that
dispatches to multiple independent sub-modules, and as a set of python
packages that may be used programatically. Where possible,
computationally-intensive algorithms are parallelized
automatically--for the ACF and the SLK steps described below--this
results in a speed-up that is linear with the number of cores available.
In all cases the implementation has been tuned to minimize memory-use and
computation time. All programs expect files in simple BED format \citep{Kent2002}
sorted by chromosome and start.
When run without arguments, the \textit{comb-p} executable displays the
available programs and a short description of each:
\begin{verbatim}
acf      - autocorrelation within BED file
slk      - Stouffer-Liptak-Kechris correction
fdr      - Benjamini-Hochberg correction
peaks    - find peaks in a BED file.
region_p - SLK p-values for a region
hist     - histogram of a p-values
splot    - scatter plot of a region.
manhattan - a manhattan plot
pipeline  - slk, fdr, peaks, region_p
\end{verbatim}
Though each of these tools may be used independently, the progression of
\textit{acf}, \textit{slk}, \textit{fdr}, \textit{peaks}, \textit{region\_p}
described below works well for the examples presented. These steps can
be run successively with the single command: \textit{pipeline}.

Given a set of auto-correlated p-values in BED format, \textit{comb-p} first
calculates the correlation at varying distance lags. Whereas Kechris' \citep{Kechris2010}
and many ACF implementations rely on fixed offsets of adjacent probes,
\textit{comb-p} accepts a maxiumum base-pair distance and a bin size
for pairing of probes based on their distance, so a given probe may
have multiple probes in each of a set of bins. This is useful in cases
where the probes generating the p-values are unevenly spaced. We refer
to this set of lagged correlations as the ACF for auto-correlation
function.

Once the ACF has been calculated, it can be used to perform the
Stouffer-Liptak-Kechris correction (\textit{slk}) where each p-value is
adjusted according to the adjacent p-values and the weight of each of
those neighboring probes is determine from the ACF. The resulting BED file
has the same number of rows with an additional column containing the corrected
p-value. A given p-value will be pulled lower if its neighbors also have low
p-values (and little autocorrelation) and likely remain insignificant if the
neighboring p-values are also high.

A q-value score based on the Benjamini-Hochberg false-discovery (FDR) correction
 may be calculated. The peak-finding algorithm can then be used on either the FDR
q-value, the \textit{slk}-corrected p-value, or on the original p-value.
We find that performing the \textit{slk} correction and the FDR correction
before finding regions is the most stable and powerful.

A p-value for each region can be assigned using the Stouffer-Liptak correction.
This step varies from the \textit{slk} step in that the \textit{slk} step uses
the ACF calculated out to a fixed lag where as the \textit{region\_p} step first
calculates the ACF out to a distance equal to the largest region-size. The
corrected p-value for each region is then calculated using the original p-values that
fall within the region, and the portion of the ACF that extends to the distance
spanned by that region. Because we use the original, uncorrected p-values in the
calculation of significance for the region, we side-step issues of altering the
distribution in both the \textit{slk} and FDR steps. The \textit{region\_p}
program reports the \textit{slk} corrected p-value, and a further one-step
\cite{Sidak}
multiple-testing correction. For a given region, the number of possible
tests in the Sidak correction is the total bases covered by all input
probes divided by the size of the given region.

\section{Application}
We extend 3 previously published analyses to demonstrate the utility and
breadth of use of \textit{comb-p} and to validate against the
published data. The full set of commands used to run each of these analyses
is availalable in the corresponding subdirectory at:
https://github.com/brentp/combined-pvalues/tree/master/examples

Comprehensive, High-throughput Array for Relative Methylation or CHARM is a
tiling array used to quantify methylation at CpG-rich sites \citep{Irizarry2008}.
Adjacent probes are spatially auto-correlated due to the regional nature of
methylation, and to the overlapping probes on the tiling array.
We run \textit{comp-p} on data from
\cite{Irizarry2009} containing probe data for tumor and normal samples from
a variety of tissues. \textit{comb-p} finds 85\% of the
tumor-specific differentially methylated regions (DMR's) reported by
\cite{Irizarry2009} and 75\% of the tissue-specific DMR's reported.
While these overlaps depend on the chosen parameters, we show that the Irizarry
DMR's that overlap with the \textit{comb-p} DMR's have signficantly (t-test,
p=1.296e-145) lower FDR values (as reported by Irizarry) than those that do not,
indicating that \textit{comb-p} is finding the most significant DMR's.

Bisulfite-sequencing is also used to measure
methylation across the genome.
We analyze data described in \cite{Hsieh2009} from {\it Arabidopsis
 thaliana}. We use MethylCoder \citep{Pedersen2011} to map the converted reads
to the genome. In comparing endosperm to
embryo, we find differentially methylated regions associated with genes
enriched for gene ontologies related to the ribosome (p < 1e-3).

DamID technology can be used to measure protein binding sites \cite{Steensel2001}.
We recreate analysis of \cite{Kechris2010}, using the p-values from their moderated
t-statistics. Of Kechris' 878 regions, 695 (79\%) are represented in the comb-p regions.
The Kechris regions that overlap the \textit{comb-p} regions have a higher reported
enrichment [TODO Katerina, is it correct to say it's an enrichment score?] score than
those that do not (t-test, p=2.98e-59), indicating that \textit{comb-p} is finding the
best regions among those reported by Kechris.

\end{methods}

\section{Conclusion}
The \textit{comb-p} software is useful in contexts where there is
autocorrelation among nearby p-values. We have outlined our implementation
and demonstrated the utility on data from three different technologies,
each from a different statistical test. We have also validated our method
using previously published results from those same studies.

\section*{Acknowledgement}

\paragraph{Funding\textcolon} This work was supported by ??? to ???.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%\bibliographystyle{plain}
%\bibliography{Document}

\begin{thebibliography}{}
\bibitem[Kechris \textit{et~al}., 2010]{Kechris2010}
Kechris, K.J. et al. (2010)
Generalizing moving averages for tiling arrays using combined p-value
statistic. {\it Statistical Applications in Genetics and Molecular Biology}
{\bf 9}, Article 29.

\bibitem[Kent \textit{et al.}, 2002]{Kent2002} Kent, W.J. et al. (2002) The human genome browser at UCSC. {\it Genome Res.}, {\bf 12}, 996-1106.

\bibitem[Sidak, 1967]{Sidak} 
Šidàk, Z. (1967).
Rectangular conﬁdence region for the means of multivariate normal distributions.
\textit{Journal of the American Statistical Association}, {\bf 62}, 626-633.

\bibitem[Fisher, 1948]{Fisher} 
Fisher, R.A. (1948)
Questions and answers \#14.
{\it The American Statistician}, {\bf 2}(5), 30-31.

\bibitem[Stouffer \textit{et al.}, 1949]{Stouffer}
Stouffer, S.A. \textit{et~al.} (1949). \textit{The American Soldier},
Princeton University Press, Princeton, NJ. Vol.1: Adjustment during Army Life.

\bibitem[Liptak, 1958]{Liptak}
Liptak, T. (1958). On the combination of independent tests. {\it Magyar Tud.
Akad. Mat. Kutato Int. Kozl.}, {\bf 3}, 171-197.

\bibitem[Zaykin {\it et~al}., 2002]{Zaykin2002} Zaykin, D.L. et al. (2002).
Truncated product method for combining p-values. {\it Genetic Epidemiology},
{\bf 22}, 170-185.

\bibitem[Irizarry {\it et~al}., 2008]{Irizarry2008} Irizarry, R.A. et al.
(2008). Comprehensive high-throughput arrays for relative methylation (CHARM).
{\it Genome Research}, {\bf 18}, 780-790.

\bibitem[Irizarry {\it et~al}., 2009]{Irizarry2009} Irizarry, R.A. et al.
(2009). The human colon cancer methylome shows similar hypo- and
hypermethylation at conserved tissue-specific CpG island shores.
{\it Nature Genetics}, {\bf 41}, 178-186.

\bibitem[Pedersen {\it et~al}., 2011]{Pedersen2011} Pedersen, B.S. et al.
(2011). MethylCoder: software pipeline for bisulfite-treated sequences.
{\it Bioinformatics}, {\bf 27}, 2435-2436.

\bibitem[van Steensel {\it et~al}., 2001]{Steensel2001}van Steensel, B., Delrow, J. and Henikoff, S. (2001): Chromatin Profiling Using Targeted DNA Adenine Methyltransferase,
{\it Nature Genetics}, {\bf 27}, 304–308.

\bibitem[Hsieh {\it et~al}., 2009]{Hsieh2009} Hsieh, T.F. et al. (2009). Genome-Wide Demethylation of Arabidopsis Endosperm.
{\it Science}, {\bf 324}, 1451-1454.

\end{thebibliography}
\end{document}
