\documentclass{bioinfo}
\copyrightyear{2012}
\pubyear{2012}

\begin{document}
\firstpage{1}

\title[comb-p]{Comb-p: A tool for combining spatially correlated p-values and assigning significance}
\author[Pedersen \textit{et~al}]{Brent S. Pedersen\,$^{1,*}$, Katerina J. Kechris\,$^{2}$,
    Ivana V. Yang\,$^{1}$ and David A. Schwartz\,$^1$}
\address{$^{1}$Department of Medicine, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
$^{2}$Department of Statistics, University of Colorado, Denver, Anschutz Medical Campus, Aurora CO 80045, USA\\
}
\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}
\editor{Associate Editor: XXXXXXX}
\maketitle
\begin{abstract}

\section{Summary:}
\textit{comb-p} is set of command-line tools and a python library that
manipulate BED files of p-values and 1) calculate autocorrelation, 2) Combine
adjacent p-values based on the Stouffer-Liptak-Kechris correction \citep{Kechris2010},
3) perform false discovery correction, 4) find troughs of low p-values, and 5)
assign significance to those regions.
In addition, a number of tools are provided for visualization and
assessment. The library provides sensible defaults for parameters at each
step based on the structure of the input data, but these may also be customized
by the user. We provide example uses on bisulfite-seq, ChIP-seq and tiled methylation
probes.

\section{Availability:}
 \textit{comb-p} is maintained under the BSD license. The documentation and
 implementation are available at

 \href{https://github.com/brentp/combined-pvalues}{https://github.com/brentp/combined-pvalues}.
\section{Contact:} \href{bpederse@gmail.com}{bpederse@gmail.com}
\end{abstract}

\section{Introduction}
ChIP-seq, bisulfite-seq, and tiling arrays generate data that is spatially
auto-correlated. These technologies and many others generate values that can be
compared or tested to generate p-values that are also auto-correlated. Due to the
high-throughput nature of these methods, the significance of individual regions
may be obscured after multiple-testing correction on potentially millions of sites.
Here, we build on the approach of combining p-values described in \citep{Kechris2010}.

\section{Approach}

\citep{Fisher} developed an approach of combining p-values from independent tests
to get a single meta-analysis test statistic with a chi-squared distribution 
and degrees of freedom based on the number of tests being combined.
A similar method developed by \citep{Stouffer} first converts p-values
to Z-scores which are then summed and scaled to create a single, combined Z-score.
Stouffer's method lends itself well to the addition of weights on each p-value.

\citep{Zaykin} extended the weighting explicitly to deal with auto-correlation and
\citep{Kechris2010} used that framework to create a sliding window correction where
the p-value of each probe is adjusted by applying Stouffer's method to neighboring
p-values as weighted according to the auto-correlation at their distance.


\begin{methods}


\section{Application}
\textit{comb-p} is implemented as a single command-line application that
dispatches to multiple independent sub-modules, and as a set of python
packages. Where possible, computationally-intensive parts of the algorithm
are parallelized automatically--for the ACF and the SLK steps described below.
This results in a speedup that is linear with the number of cores available.
When run without arguments, the \textit{comb-p} executable displays the
available programs and a short description:
\begin{verbatim}
acf      - autocorrelation within BED file
slk      - Stouffer-Liptak-Kechris correction
fdr      - Benjamini-Hochberg correction
peaks    - find peaks in a BED file.
region_p - SLK p-values for a region
hist     - histogram of a p-values
splot    - scatter plot of a region.
manhattan - a manhattan plot
\end{verbatim}

Given a set of auto-correlated p-values in BED format, \textit{comb-p} first
calculates the correlation at varying distance lags. Whereas \citep{Kechris2010}
and many ACF implementations rely on fixed offsets of adjacent probes,
\textit{comb-p} accepts a maxiumum base-pair distance and bins pairs of probes
based on their distance, so a given probe may have multiple probes in each of
a set of bins. This is useful in cases where the probes generating the p-values
are unevenly spaced. We refer to this set of lagged correlations as the ACF for
auto-correlation function.

Once the ACF has been calculated, it can be used to perform the
Stouffer-Liptak-Kechris correction (\textit{slk}) where each p-value is
adjusted according to the p-values in its neighborhood and the weight applied
to each probe that neighborhood according to the ACF. The resulting BED file
has the same number of rows with an additional column containing the corrected
p-value. A given p-value will be pulled lower if it's neighbors also have low
p-values and likely remain insignificant if the neighboring p-values are also 
high.

A q-value score based on the Benjamini-Hochberg false-discovery (FDR) correction
 may be calculated. A peak-finding algorithm can then be used on either the FDR
q-value, the \textit{slk}-corrected p

\end{methods}

\section{Conclusion}
The \textit{comb-p} software is useful in contexts where there is autocorrelation among nearby p-values.

\section*{Acknowledgement}

\paragraph{Funding\textcolon} This work was supported by ???.

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}

%\bibliographystyle{plain}

%\bibliography{Document}
\begin{thebibliography}{}
\bibitem[Kechris {\it et~al}., 2010]{Kechris2010} Kechris, K.J. et al. (2010).
Generalizing moving averages for tiling arrays using combined p-value
statistic. {\it Statistical Applications in Genetics and Molecular Biology}
{\bf 9}, Article 29.
\bibitem[Quinlan and Hall, 2010]{Quinlan2010} Quinlan, A.R. and Hall, I.M. (2010) BEDTools: a flexible suite of utilities for comparing genomic features, {\it Bioinformatics}, {\bf 26}, 841-842.
\bibitem[Kent \textit{et al.}, 2002]{Kent2002} Kent, W.J. et al. (2002) The human genome browser at UCSC. {\it Genome Res.}, {\bf 12}, 996-1106.

\end{thebibliography}
\end{document}
